rs/nervous_system/integration_tests/tests/upgrade_sns_controlled_canister_with_large_wasm.rs

```diff
@@ -0,0 +1,218 @@
+    use super::*;
+
+    use candid::{Decode, Encode};
+    use pocket_ic::nonblocking::PocketIc;
+    use pocket_ic::WasmResult;
+
+    /// Interim test function for calling Root.change_canister.
+    ///
+    /// This function is not in src/pocket_ic_helpers.rs because it's going to be replaced with
+    /// a proposal with the same effect. It should not be used in any other tests.
+    pub async fn change_canister(
+        pocket_ic: &PocketIc,
+        canister_id: PrincipalId,
+        sender: PrincipalId,
+        request: ChangeCanisterRequest,
+    ) {
+        let result = pocket_ic
+            .update_call(
+                canister_id.into(),
+                sender.into(),
+                "change_canister",
+                Encode!(&request).unwrap(),
+            )
+            .await
+            .unwrap();
+        let result = match result {
+            WasmResult::Reply(result) => result,
+            WasmResult::Reject(s) => panic!("Call to change_canister failed: {:#?}", s),
+        };
+        Decode!(&result, ()).unwrap()
+    }
+}
+
+async fn run_test(store_same_as_target: bool) {
+    // 1. Prepare the world
+    let pocket_ic = PocketIcBuilder::new()
+        .with_nns_subnet()
+        .with_sns_subnet()
+        .with_application_subnet()
+        .build_async()
+        .await;
+
+    // Install the NNS canisters.
+    {
+        let with_mainnet_nns_canisters = false;
+        install_nns_canisters(&pocket_ic, vec![], with_mainnet_nns_canisters, None, vec![]).await;
+    }
+
+    // Publish SNS Wasms to SNS-W.
+    {
+        let with_mainnet_sns_canisters = false;
+        add_wasms_to_sns_wasm(&pocket_ic, with_mainnet_sns_canisters)
+            .await
+            .unwrap();
+    };
+
+    // Install a dapp canister.
+    let original_wasm = Wasm::from_bytes(UNIVERSAL_CANISTER_WASM.to_vec());
+    let original_wasm_hash = original_wasm.sha256_hash();
+
+    let app_subnet = pocket_ic.topology().await.get_app_subnets()[0];
+
+    let target_canister_id = install_canister_on_subnet(
+        &pocket_ic,
+        app_subnet,
+        vec![],
+        Some(original_wasm.clone()),
+        vec![ROOT_CANISTER_ID.into()],
+    )
+    .await;
+
+    let sns = {
+        let create_service_nervous_system = CreateServiceNervousSystemBuilder::default()
+            .with_dapp_canisters(vec![target_canister_id])
+            .build();
+
+        let swap_parameters = create_service_nervous_system
+            .swap_parameters
+            .clone()
+            .unwrap();
+
+        let sns_instance_label = "1";
+        let (sns, _) = nns::governance::propose_to_deploy_sns_and_wait(
+            &pocket_ic,
+            create_service_nervous_system,
+            sns_instance_label,
+        )
+        .await;
+
+        sns::swap::await_swap_lifecycle(&pocket_ic, sns.swap.canister_id, Lifecycle::Open)
+            .await
+            .unwrap();
+        sns::swap::smoke_test_participate_and_finalize(
```

Comment:
It would be better to not have all our tests do this.  This was originally created for the sake of final release qualification and is the slowest part of most tests.  

PocketIC tests seem to be a bit slower for some reason